spec/forms/inheritance_form_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+RSpec.describe AdminForm do
+  context 'class methods' do
+    describe '.properties' do
+      it 'returns an array with email and role' do
+        expect(AdminForm.properties).to contain_exactly(:email, :role, :permissions)
+      end
+    end
+  end
+
+  context 'instance methods' do
+    let(:user) { build :user }
+    subject { described_class.new(user) }
+
+    describe 'accessors for properties' do
+      it 'allows reading and writing for email' do
+        subject.email = 'admin@example.com'
+
+        expect(subject.email).to eq 'admin@example.com'
+      end
+
+      it 'allows reading and writing for role' do
+        subject.role = 'superadmin'
+
+        expect(subject.role).to eq 'superadmin'
+      end
+    end
+
+    describe 'accessors for AdminForm specific properties' do
+      it 'allows reading and writing for permissions' do
+        subject.permissions = 'edit_users'
+
+        expect(subject.permissions).to eq 'edit_users'
+      end
+    end
```

Comment:
nitpick: personally, I would merge all tests about specific fields into one, like this:
 
 ```ruby
      let(:fields) { [:email, :role, :permissions] } 
      
      describe 'properties field' do
        it 'returns an array with email and role' do
          expect(AdminForm.properties).to contain_exactly(fields)
        end
      end

      describe 'setting up values' do
        it 'sets up values' do
          fields.map do |field|
            subject.send(field) = 'somestr'
            expect(subject.send(field)).to eq 'somestr'
          end
        end
      end
```

as separate testing of specific fields gives not much of a value.